src/commands/ingress/tryConfigureIngressUsingDockerfile.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzExtFsExtra } from '@microsoft/vscode-azext-utils';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../utils/localize';
+import type { IngressContext } from './IngressContext';
+import { getDefaultPort } from './editTargetPort/getDefaultPort';
+
+export async function tryConfigureIngressUsingDockerfile(context: IngressContext): Promise<void> {
+    if (!context.dockerfilePath) {
+        return;
+    }
+
+    context.dockerfileExposePorts = await getDockerfileExposePort(context.dockerfilePath);
+
+    if (context.alwaysPromptIngress) {
+        return;
+    }
+
+    if (!context.dockerfileExposePorts) {
+        context.enableIngress = false;
+        context.enableExternal = false;
+    } else if (context.dockerfileExposePorts) {
+        context.enableIngress = true;
+        context.enableExternal = true;
+        context.targetPort = getDefaultPort(context);
+    }
+
+    // If a container app already exists, activity children will be added automatically in later execute steps
```

Comment:
Will add this back in once we add all the activity children support in upcoming PR